File: frontend/src/services/authService.js
import { apiClient, storeTokens } from "@/lib/apiClient"

export const authService = {
  async login(email, password) {
    const { data } = await apiClient.post("/auth/token/", { email, password })
    storeTokens({ access: data.access, refresh: data.refresh })
    return data.user
  },

  async register(payload) {
    const { data } = await apiClient.post("/auth/register/", payload)
    return data
  },

  async registerEcole(payload) {
    const { data } = await apiClient.post("/auth/register/ecole/", payload)
    return data
  },

  async fetchProfile() {
    const { data } = await apiClient.get("/auth/profile/")
    return data
  },

  async updateProfile(payload) {
    const { data } = await apiClient.patch("/auth/profile/", payload)
    return data
  },

  async changePassword(ancienMotDePasse, nouveauMotDePasse) {
    const { data } = await apiClient.post("/auth/changer-mot-de-passe/", {
      ancien_mot_de_passe: ancienMotDePasse,
      nouveau_mot_de_passe: nouveauMotDePasse,
    })
    return data
  },

  logout() {
    storeTokens(null)
  },
}

File: cross/lib/core/auth/auth_service.dart
import '../api/api_client.dart';
import '../api/token_storage.dart';
import '../../models/user.dart';

/// Miroir de `authService.js` — mêmes endpoints, même contrat (`{access, refresh, user}`).
class AuthService {
  final _dio = ApiClient.instance.dio;

  /// [identifiant] accepte un email OU un matricule (résolu côté backend, voir
  /// `CustomTokenObtainPairSerializer`).
  Future<AppUser> login(String identifiant, String password) async {
    final response = await _dio.post('/auth/token/', data: {
      'email': identifiant,
      'password': password,
    });
    final data = response.data as Map<String, dynamic>;
    await TokenStorage.instance.save(access: data['access'] as String, refresh: data['refresh'] as String);
    return AppUser.fromJson(data['user'] as Map<String, dynamic>);
  }

  Future<AppUser> fetchProfile() async {
    final response = await _dio.get('/auth/profile/');
    return AppUser.fromJson(response.data as Map<String, dynamic>);
  }

  Future<void> changePassword(String ancien, String nouveau) async {
    await _dio.post('/auth/changer-mot-de-passe/', data: {
      'ancien_mot_de_passe': ancien,
      'nouveau_mot_de_passe': nouveau,
    });
  }

  Future<void> logout() async {
    await TokenStorage.instance.clear();
  }
}
